lib/aerospike.js

```diff
@@ -235,62 +234,413 @@ var prepend = function prepend (key, bins, metadata, policy, callback) {
 // `aerospike` shim
 // ****************************************************************************
 
-var aerospike = {
-  key: as.key,
-  status: as.status,
-  policy: as.policy,
-  language: as.language,
-  log: as.log,
-  scanPriority: as.scanPriority,
-  predicates: as.predicates,
-  indexType: as.indexType,
-  scanStatus: as.scanStatus,
-  Double: as.Double
-}
-
-aerospike.client = function client (config) {
-  var client = as.client(config)
+function Aerospike () {
+  this.key = as.key,
+  this.status = as.status,
+  this.status_ok = this.status.AEROSPIKE_OK,
+  this.policy = as.policy,
+  this.language = as.language,
+  this.log = as.log,
+  this.scanPriority = as.scanPriority,
+  this.predicates = as.predicates,
+  this.indexType = as.indexType,
+  this.scanStatus = as.scanStatus,
+  this.Double = as.Double
+
+  this._currentClient = {}
+  this.proto = {}
+}
+
+Aerospike.prototype._client = function client (config) {
+  let client = as.client(config)
   if (client === null) {
     throw new Error('Client object creation failed - null value returned')
   }
 
-  var proto = Object.getPrototypeOf(client)
-  if (!proto.createQuery) {
-    proto.createQuery = proto.query
-    proto.query = query
-  }
+  this.proto = Object.getPrototypeOf(client)
```

Comment:
This is just wrapped what was before. I am sure this will disappear, but I haven't touched it yet. Further down you'll see `return client.`. This would be the mechanism to wrap the client in a `this._currentClient` in order to do two things:
* wrapping client to handle errors in class
* combining client creation and `client.connect()`